### PR TITLE
[RDY] Batch annotations

### PR DESCRIPTION
**Pull request by gfokkema**

### CODE REVIEW

File: dnainator-core/src/main/java/nl/tudelft/dnainator/core/impl/Cluster.java [hunk 1]
```diff
@@ -10,20 +11,31 @@
...
 	 */
-	public Cluster(int rankStart, List<SequenceNode> list) {
+	public Cluster(int rankStart, List<SequenceNode> list, List<Annotation> annotations) {
 		this.rankStart = rankStart;
+		this.annotations = annotations;
 		this.nodes = list;
 	}
 
 	/**
+	 * Test.
```
Comment (Unia) [inline dnainator-core/src/main/java/nl/tudelft/dnainator/core/impl/Cluster.java]: You probably want to replace this placeholder :stuck_out_tongue:

File: dnainator-core/src/main/java/nl/tudelft/dnainator/graph/impl/Neo4jGraph.java [hunk 2]
```diff
@@ -197,4 +205,52 @@ public void execute(Command c) {
...
+			Node annotation = e.createNode(NodeLabels.ANNOTATION);
+			annotation.setProperty(PropertyTypes.ID.name(), a.getGeneName());
+			annotation.setProperty(PropertyTypes.STARTREF.name(), a.getStart());
+			annotation.setProperty(PropertyTypes.ENDREF.name(), a.getEnd());
+			annotation.setProperty(PropertyTypes.SENSE.name(), a.isSense());
+
+			ResourceIterator<Node> nodes = service.execute(GET_RANGE, parameters).columnAs("n");
+			nodes.forEachRemaining(n -> n.createRelationshipTo(annotation, RelTypes.ANNOTATED));
+		});
+		System.out.println("Finished loading annotations.");
```
Comment (Unia) [inline dnainator-core/src/main/java/nl/tudelft/dnainator/graph/impl/Neo4jGraph.java]: Still needed?

File: dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/drawables/strains/ClusterDrawable.java [hunk 3]
```diff
@@ -62,6 +63,13 @@ private void draw(ColorServer colorServer) {
 			getChildren().add(pie);
 		}
 		getChildren().add(label);
+		// TODO: Uncomment everything below to draw annotations. VERY SLOW!
```
Comment (Unia) [inline dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/drawables/strains/ClusterDrawable.java]: Can be removed.

**Comment by Unia** [comment]
Besides nitpicks, can be merged :)

File: dnainator-core/src/main/java/nl/tudelft/dnainator/graph/impl/Neo4jGraph.java (hunk 2 above)
Comment (gfokkema) [inline dnainator-core/src/main/java/nl/tudelft/dnainator/graph/impl/Neo4jGraph.java]: The Graph interface now extends the AnnotationCollection interface, therefore it has to implement all annotation methods.

A nice extra is that theoretically it is still possible to first load the graph and then the annotations. It's just terribly slow.

Comment (Unia) [inline dnainator-core/src/main/java/nl/tudelft/dnainator/graph/impl/Neo4jGraph.java]: I was aiming for the println ;)

**Comment by Unia** [comment]
Merging

**Pull request merged**